gems/smithy/lib/smithy/generators/schema.rb

```diff
@@ -11,26 +11,47 @@ def initialize(plan)
         super
       end
 
-      # @return [Enumerator<String, String>] The file paths and their contents to generate.
       def generate
-        files = source_files
-        files.each do |file, content|
+        gem_files.each_with_object([]) do |(file, content), files|
+          next if file == "lib/#{@gem_name}/customizations.rb" && should_skip_customizations?
+
           create_file file, content
+          files << file
         end
-        files
+      end
+
+      def source
+        source = source_files.map { |_file, content| content }.join("\n")
+        Object.module_eval(source)
+        Object.const_get(@plan.gem_namespace)
+      rescue Exception => e # rubocop:disable Lint/RescueException
+        puts "Error evaluating source:\n#{source}"
```

Comment:
Wouldn't the source be large here?

Reply:
Possibly, but not as large as you would think, and I think the size is/will be the same with the current v3. If you're talking about the debugging, yeah, I think in a future change I plan to add a logger and this can be a debug (opt-in).